src/lib/parse/blueprint.js

```diff
@@ -28,7 +41,36 @@ module.exports = function(filePath, autoOptions, routeMap) {
 
             cb();
 
-            function setupResourceAndUrl(resource) {
+            function validateAndSetupResource(fixtureResource: BlueprintResource, contract: Contract) {
+                const fixtureUrl = urlParser.parse(fixtureResource.uriTemplate).url;
+               
+                const contractActions: ?Actions = getActions(fixtureUrl, contract.resources);
+
+                let valdatedResource = contracts.removeInvalidActions(fixtureResource, contractActions);
+                if (!valdatedResource.actions.length) {
+                    return;
+                }
+                var parsedUrl = urlParser.parse(valdatedResource.uriTemplate);
+                var key = parsedUrl.url;
+                routeMap[key] = routeMap[key] || { urlExpression: key, methods: {} };
+                parseParameters(parsedUrl, valdatedResource.parameters, routeMap);
+                valdatedResource.actions.forEach(function(action){
+                    parseAction(parsedUrl, action, routeMap);
+                    saveRouteToTheList(parsedUrl, action);
+                });
+            }
+
+            function getActions(fixtureUrl: string, contractResources: Resources): ?Actions {
```

Comment:
add comment: `contractResources` is expected to already be sorted in match-priority order.